File: double_dispatch/java/02_double_dispatch/Scissor.java
public class Scissor implements Beatable {
	public boolean beats(Beatable other) {
		return other.loseAgainst(this);
	}

	@Override
	public boolean loseAgainst(Rock rock) {
		return true;
	}

	@Override
	public boolean loseAgainst(Paper paper) {
		return false;
	}

	@Override
	public boolean loseAgainst(Scissor scissor) {
		return false;
	}
}

File: 02_double_dispatch/c#/01_before_pattern_matching/Scissor.cs
public class Scissor : Beatable {
	public string Beats(Beatable other) {
		if (other.GetType() == typeof(Rock)) {
			return "scissor loses";
		}

		if (other.GetType() == typeof(Paper)) {
			return "scissor wins";
		}
		return "draw";
	}
}
